e2e/rbd.go

```diff
@@ -554,78 +475,32 @@ var _ = Describe("RBD", func() {
 				}
 				e2elog.Logf("rbd-nbd package version: %v", rpmv)
 
-				// Get details of rbd-nbd process
-				// # ps -eo 'cmd' | grep [r]bd-nbd
-				// /usr/bin/rbd-nbd --id cephcsi-rbd-node -m svc-name:6789 --keyfile=/tmp/csi/keys/keyfile attach \
-				// --device /dev/nbd0 pool-name/image-name --try-netlink --reattach-timeout=180
-				mapCmd, stdErr, err := execCommandInContainer(
-					f,
-					"ps -eo 'cmd' | grep [r]bd-nbd",
-					cephCSINamespace,
-					"csi-rbdplugin",
-					&opt)
-				if err != nil || stdErr != "" {
-					e2elog.Failf("failed to run ps cmd : %v, stdErr: %v ", err, stdErr)
-				}
-				e2elog.Logf("map command running before restart, mapCmd: %v", mapCmd)
-
-				rbdNodeKey, stdErr, err := execCommandInToolBoxPod(
-					f,
-					"ceph auth get-key client.cephcsi-rbd-node",
-					rookNamespace)
-				if err != nil || stdErr != "" {
-					e2elog.Failf("error getting cephcsi-rbd-node key, err: %v, stdErr: %v ", err, stdErr)
-				}
-
-				// restart the rbd node plugin
-				err = deletePodWithLabel(selector, cephCSINamespace, false)
-				if err != nil {
-					e2elog.Failf("fail to delete pod with error %v", err)
-				}
-
-				// wait for nodeplugin pods to come up
-				err = waitForDaemonSets(rbdDaemonsetName, cephCSINamespace, f.ClientSet, deployTimeout)
-				if err != nil {
-					e2elog.Failf("timeout waiting for daemonset pods with error %v", err)
-				}
-
-				// Prepare the rbd-nbd with command args
-				attachCmd := strings.ReplaceAll(mapCmd, "map", "attach --device /dev/nbd0")
-				m1 := regexp.MustCompile(`/keyfile-[0-9]* `)
-				attachCmd = m1.ReplaceAllString(attachCmd, "/keyfile-test ")
-				e2elog.Logf("attach command to run after restart, attachCmd: %v", attachCmd)
-
-				// create the keyfile
-				_, stdErr, err = execCommandInContainer(
-					f,
-					fmt.Sprintf("echo %s > /tmp/csi/keys/keyfile-test", rbdNodeKey),
-					cephCSINamespace,
-					"csi-rbdplugin",
-					&opt)
-				if err != nil || stdErr != "" {
-					e2elog.Failf("failed to write key to a file,  err: %v, stdErr: %v ", err, stdErr)
-				}
-
-				_, stdErr, err = execCommandInContainer(f, attachCmd, cephCSINamespace, "csi-rbdplugin", &opt)
-				if err != nil || stdErr != "" {
-					e2elog.Failf("failed to run attach cmd err: %v, stdErr: %v ", err, stdErr)
-				}
-
-				runningAttachCmd, stdErr, err := execCommandInContainer(
-					f,
-					"ps -eo 'cmd' | grep [r]bd-nbd",
-					cephCSINamespace,
-					"csi-rbdplugin",
-					&opt)
-				if err != nil || stdErr != "" {
-					e2elog.Failf("failed to run ps cmd : %v, stdErr: %v ", err, stdErr)
+				// Lets give the healer sometime to trigger nodeStageVolume
+				var runningAttachCmd string
+				for i := 1; i <= maxRetries; i++ {
```

Comment:
isn't it common to start counting at 0?

Reply:
isn't it only used on line 520?
```go
if i < maxRetries {
```

In general, it might be cleaner to rewrite the loop to use `wait.PollImmediate()` as we do in other places.

Reply:
521 too.

Yeah, I like the idea. Let me use it.